src/main/java/com/example/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,35 @@
+package com.example.repository.book;
+
+import com.example.dto.BookSearchParametersDto;
+import com.example.model.Book;
+import com.example.repository.SpecificationBuilder;
+import com.example.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+
+    private static final String AUTHOR = "author";
+    private static final String TITLE = "title";
```

Comment:
better make these constants public and have only one constant and one place for future changing